lib/katalyst/tables/backend/sort_form.rb

```diff
@@ -43,18 +43,30 @@ def status(column)
         # @param column [String, Symbol] the table column as defined in table_with
         # @return [String] URL for use as a link in a column header
         def url_for(column)
-          direction = "asc"
-
-          if column.to_s == self.column
-            case self.direction
-            when "asc"
-              direction = "desc"
-            when "desc"
-              direction = "asc"
-            end
-          end
+          # Implementation inspired by pagy's `pagy_url_for` helper.
+
+          request = @controller.request
 
-          @controller.url_for(sort: "#{column} #{direction}")
+          # Preserve any existing GET parameters
+          # CAUTION: these parameters are not sanitised
```

Comment:
Should we?

Reply:
I don't think that we can – we want to preserve the existing GET parameters and this approach makes sure that they are only used as GET parameters. We don't know what those parameters might be.

FYI pagy uses the same approach